### DISCUSSION TITLE
Is there a way to change provider configuration for single task?

### DISCUSSION THREAD

**Discussion by mhitza** [pulumi/pulumi#7879] [Pulumi in Practice] [answered]
I'm spinning up an AWS infrastructure project in `eu-central-1` (as an example), and one of the resources provisioned would be CloudFront with a custom domain. However in order to attach an HTTPS certificate generated by AWS Certificate Manager, CloudFront requires the certificate to be generated in the `us-east-1` region.

Is there a way to alter the `aws:region` temporarily for a group of tasks?
If not I'm curious how others have handled similar scenarios, because I can only think of either moving the entire stack to `us-east-1` (the easy way) or separating the AWS certificate generation + CloudFront in a separate project (undesired because it might involve manual tweaking between projects).

**Comment by cnunciato** [comment, marked as answer]
Apologies for the _super_ delayed response on this!

Yes, you can apply a different region by instantiating a provider in your program and then using it in resource declarations. Here's a short program, for example, that uses the stack's configured provider to create an S3 bucket in one region and an "explicit" provider (i.e., one that's local to the program) to create a second bucket in a different region:

In `Pulumi.dev.yaml`:

```yaml
config:
  aws:region: us-west-2
```

In `index.ts`:

```typescript
import * as aws from "@pulumi/aws";

// Create a bucket using the default configured provider.
const myUSWest2Bucket = new aws.s3.Bucket("my-us-west-2-bucket");

// Create a us-east-1 provider.
const myUSEast1Provider = new aws.Provider("my-us-east-1-provider", {
    region: "us-east-1",
});

// Create a bucket using the us-east-1 provider.
const myUSEast1Bucket = new aws.s3.Bucket("my-us-east-1-bucket", undefined, {
    provider: myUSEast1Provider,
});
``` 

And the result after deployment:

![image](https://user-images.githubusercontent.com/274700/157152023-033ab16c-ae96-4b42-8205-fd863d46003a.png)

More on explicit provider configuration in the Providers docs: 
https://www.pulumi.com/docs/intro/concepts/resources/providers/#explicit-provider-configuration

You can also define an explicit provider on a Component Resource, and have all of that resource's children inherit that same provider:
https://www.pulumi.com/docs/intro/concepts/resources/components/#inheriting-resource-providers
